src/utils/middlewares/azure_user_attributes_manage.ts

```diff
@@ -86,17 +101,71 @@ export const AzureUserAttributesManageMiddleware = (): IRequestMiddleware<
 
   if (subscriptionId.startsWith("MANAGE-")) {
     // MANAGE Flow
-    const authInfo: IAzureUserAttributesManage = {
-      email: userEmail,
-      kind: "IAzureUserAttributesManage"
-    };
-
-    return E.right<
-      IResponse<
-        "IResponseErrorForbiddenNotAuthorized" | "IResponseErrorInternal"
-      >,
-      IAzureUserAttributesManage
-    >(authInfo);
+    // TODO: The condition MUST be removed after io-function-services update
+    if (authorizedCIDRsModel) {
+      const errorOrMaybeAuthorizedCIDRs = await authorizedCIDRsModel.find([
+        subscriptionId
+      ])();
+
+      if (E.isLeft(errorOrMaybeAuthorizedCIDRs)) {
+        winston.error(
+          `No CIDRs found for subscription|${subscriptionId}|${JSON.stringify(
+            errorOrMaybeAuthorizedCIDRs.left
+          )}`
+        );
+        return E.left<
+          IResponse<"IResponseErrorQuery">,
+          IAzureUserAttributesManage
+        >(
+          ResponseErrorQuery(
+            `Error while retrieving CIDRs tied to the provided subscription id`,
+            errorOrMaybeAuthorizedCIDRs.left
+          )
+        );
+      }
+
+      const maybeAuthorizedCIDRs = errorOrMaybeAuthorizedCIDRs.right;
+
+      if (isNone(maybeAuthorizedCIDRs)) {
+        winston.error(
+          `IAzureUserAttributesManage|CIDRs not found|${subscriptionId}`
+        );
+        return E.left<
+          IResponse<"IResponseErrorForbiddenNotAuthorized">,
+          IAzureUserAttributesManage
+        >(ResponseErrorForbiddenNotAuthorized);
+      }
+
+      const authInfo: IAzureUserAttributesManage = {
+        authorizedCIDRs: maybeAuthorizedCIDRs.value.cidrs
+          ? maybeAuthorizedCIDRs.value.cidrs
+          : new Set((["0.0.0.0/0"] as unknown) as ReadonlyArray<CIDR>),
+        email: userEmail,
+        kind: "IAzureUserAttributesManage"
+      };
+
+      return E.right<
+        IResponse<
+          "IResponseErrorForbiddenNotAuthorized" | "IResponseErrorInternal"
+        >,
+        IAzureUserAttributesManage
+      >(authInfo);
+    } else {
+      const authInfo: IAzureUserAttributesManage = {
+        authorizedCIDRs: new Set((["0.0.0.0/0"] as unknown) as ReadonlyArray<
```

Comment:
I don't think we should make this assumption. We are converting one misconfiguration (to forget to pass a model instance) into a security flaw (we are allowing every IP).

One question we can make to ourselves is: **is it possible, in our domain, that we have a Manage Key and we cannot look up for its cirds?** If the answer is `no`, we should consider two options:

* to make such case irrepresentable 
* to fail the operation immediately

The first to be preferred.